syntax = "proto3";

package secretsharing;

// Defines the service
service SecretSharing {
    // Function invoked to send the request
    rpc GetAttestation (AttestationRequest) returns (AttestationReply) {}
    rpc DecryptAttachedData (EncryptedData) returns (Reply) {}
}

enum CommandRequest {
    ATTESTATION = 0;
    SYNC_SYM = 1;
}

// The request message containing requested numbers
message AttestationReply {
    bool ok = 1;
    string token = 2;
    string msg = 3;
}

// The request message containing requested numbers
message AttestationRequest {
    CommandRequest cmd = 1;
}

message EncryptedData {
    repeated int32 data = 1;
}

message Reply {
    bool ok = 1;
    string msg = 2;
}
